server/http.go

```diff
@@ -298,3 +234,43 @@ func splitInstancePath(route string) (instanceURL string, remainingPath string)
 	}
 	return string(id), leadingSlash + strings.Join(ss[2:], "/")
 }
+
+func (p *Plugin) handleResponse(fn func(w http.ResponseWriter, r *http.Request) (int, error)) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		status, err := fn(w, r)
+		switch {
+		case err == nil && status == http.StatusOK:
+			p.API.LogDebug("OK: ", "Status", strconv.Itoa(status), "Path", r.URL.Path, "Method", r.Method, "query", r.URL.Query().Encode())
+		case status == 0:
+			p.API.LogDebug("Passed to another router: ", "Path", r.URL.Path, "Method", r.Method)
+		case err != nil:
+			p.API.LogError("ERROR: ", "Status", strconv.Itoa(status), "Error", err.Error(), "Path", r.URL.Path, "Method", r.Method, "query", r.URL.Query().Encode())
+		default:
+			p.API.LogDebug("unexpected plugin response", "Status", strconv.Itoa(status), "Path", r.URL.Path, "Method", r.Method, "query", r.URL.Query().Encode())
+		}
+	}
+}
+
+func (p *Plugin) handleResponseWithCallbackInstance(fn func(w http.ResponseWriter, r *http.Request, instanceID types.ID) (int, error)) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		instanceURL, _ := splitInstancePath(r.URL.Path)
+
+		callbackInstanceID, err := p.ResolveWebhookInstanceURL(instanceURL)
+		if err != nil {
+			_, _ = respondErr(w, http.StatusInternalServerError, err)
+			return
+		}
+
+		status, err := fn(w, r, callbackInstanceID)
```

Comment:
@hanzei, can you please elaborate?

Reply:
By replacing this line and the following with
```suggestion
	p.handleResponse(func(w http.ResponseWriter, r *http.Request) (int, error) {
		return fn(w, r, callbackInstanceID)
	})
```

we estimate the duplicate code that takes care of the status code..

Reply:
@hanzei the suggested logic would not work, as the underlying functions being called were not getting the expected callbackInstanceID. Hence instead of this, I have now added a common function to parse the status and the error and log them from a single function instead of writing duplicate code. Hope this is fine. 

https://github.com/mattermost/mattermost-plugin-jira/pull/923/commits/b6ff1d842b78cc8e4830627a4dba3bdb478637df